### PR TITLE
#5726 Improve encrypted message detection for multipart/encrypted email

### LINKED ISSUE #5726: Public key-encrypted message named after "noname" didn't recognized as encrypted message based on its content
### Description

We received a report send to `flowcrypt compatibility` account about a Public key-encrypted message attachment named after "noname" and another "noname" file where it contains `Version: 1`. The application used the [Canary Mail](https://canarymail.io/blog/pgp-encryption/).

The issue produces a seemingly blank page:

image

The `noname` file attachment content:

image

Reference: https://mail.google.com/mail/u/flowcrypt.compatibility@gmail.com/#inbox/FMfcgzGxTFXMFnDNbgknpqvXKtSmMgSK

### PR DESCRIPTION
**Pull request by martgil**
This PR Improves detection for an `multipart/encrypted` emails with `application/pgp-encrypted` parts.

close #5726

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated - <strike>I have asked the sender of the test email if we can incorporate the sent test email to our codebase.</strike>

Edit: I have been able to replace the sensitive part of the email while still preserving the main email structure of a multipart/encrypted email so asking for the actual email is not necessary needed.

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Comment by martgil** [comment]
Hello @sosnovsky, this is now ready for a review. 

It's also been helpful if we could share a build for him. But generally speaking, the bug is fixed and the "noname" multipart/encrypted where successfully recognized.

image

I'll be crafting a custom build and share it with them once I have your approval. Thank you!

File: extension/js/common/api/email-provider/gmail/gmail-parser.ts
Comment (sosnovsky) [inline extension/js/common/api/email-provider/gmail/gmail-parser.ts:170]: here let's check if mimeType is equal to `application/pgp-encrypted`, not just includes, so it'll look like this:
```ts
... || parts[0].mimeType === 'application/pgp-encrypted'
```

**Review by sosnovsky — COMMENTED** [review]
works well 👍

File: extension/js/common/api/email-provider/gmail/gmail-parser.ts
Comment (martgil) [reply extension/js/common/api/email-provider/gmail/gmail-parser.ts]: Thanks, Roma - I'll update the checking with a more reliable one. By the way, looks like similar scenario happens on the iOS app where the `multipart/encrypted` with encrypted emails are not recognizable. While I can't promise I can fix the same issue there, I'm also interested in doing so. I'll be checking in android app too if they its capable of rendering such email and file GH issue regarding the issue if the issue exists there too.

Comment (martgil) [reply extension/js/common/api/email-provider/gmail/gmail-parser.ts]: This issue doesn't exists on the FlowCrypt android app and they have similar detection made: https://github.com/FlowCrypt/flowcrypt-android/blob/e58d040f8eade675ca8c6bd18cb608d46bcdb44b/FlowCrypt/src/main/java/com/flowcrypt/email/extensions/jakarta/mail/PartExt.kt#L82

Comment (sosnovsky) [reply extension/js/common/api/email-provider/gmail/gmail-parser.ts]: @martgil I assigned iOS issue for you, but if you'll have any issues implementing the fix - please let me know

**Review by sosnovsky — APPROVED** [review]
👍

File: extension/js/common/api/email-provider/gmail/gmail-parser.ts
Comment (martgil) [reply extension/js/common/api/email-provider/gmail/gmail-parser.ts]: Thanks for the opportunity to familiarize myself with the iOS app's codebase. I'll explore it and let you know if I have any questions.

**Pull request merged**
